lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -278,10 +278,15 @@ def initconfig_hardware_entries(self):
             entries.append("# ROCm")
             entries.append("#------------------{0}\n".format("-" * 30))
 
-            if spec.satisfies("^blt@0.7:"):
-                rocm_root = os.path.dirname(spec["llvm-amdgpu"].prefix)
-                entries.append(cmake_cache_path("ROCM_PATH", rocm_root))
-            else:
+            rocm_root = os.path.dirname(spec["llvm-amdgpu"].prefix)
+            entries.append(cmake_cache_path("ROCM_PATH", rocm_root))
+
+            archs = self.spec.variants["amdgpu_target"].value
+            if archs[0] != "none":
+                arch_str = ";".join(archs)
+                entries.append(cmake_cache_string("CMAKE_HIP_ARCHITECTURES", arch_str))
+
+            if spec.satisfies("^blt@:0.6"):
```

Comment:
What about packages that don't use `blt` at all.

Reply:
For those, the proposal here is to do the same as for projects using BLT@0.7: